modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/ImportObjectDefinitionMVCActionCommand.java

```diff
@@ -117,44 +118,97 @@ private void _importObjectDefinition(ActionRequest actionRequest)
 
-		objectDefinition.setActive(false);
+		for (Object object : jsonArray) {
+			JSONObject jsonObject = (JSONObject)object;
 
-		String externalReferenceCode = ParamUtil.getString(
-			actionRequest, "externalReferenceCode");
+			jsonObject.remove("accountEntryRestrictedObjectFieldId");
 
-		if (Validator.isNotNull(externalReferenceCode)) {
-			objectDefinition.setExternalReferenceCode(externalReferenceCode);
-		}
+			ObjectDefinition objectDefinition = ObjectDefinition.toDTO(
+				jsonObject.toString());
+
+			objectDefinition.setActive(false);
+
+			String externalReferenceCode = ParamUtil.getString(
+				actionRequest, "externalReferenceCode");
+
+			if (Validator.isNotNull(externalReferenceCode)) {
+				objectDefinition.setExternalReferenceCode(
+					externalReferenceCode);
+			}
+
+			String name = ParamUtil.getString(actionRequest, "name");
+
+			if (Validator.isNotNull(name)) {
+				objectDefinition.setName(name);
+			}
+
+			objectDefinition.setObjectFolderExternalReferenceCode(
+				ParamUtil.getString(
+					actionRequest, "objectFolderExternalReferenceCode"));
+
+			try {
+				ObjectDefinition putObjectDefinition =
+					objectDefinitionResource.
+						putObjectDefinitionByExternalReferenceCode(
+							objectDefinition.getExternalReferenceCode(),
+							objectDefinition);
+
+				putObjectDefinition.setPortlet(objectDefinition.getPortlet());
 
-		objectDefinition.setName(ParamUtil.getString(actionRequest, "name"));
-		objectDefinition.setObjectFolderExternalReferenceCode(
-			ParamUtil.getString(
-				actionRequest, "objectFolderExternalReferenceCode"));
+				if (!FeatureFlagManagerUtil.isEnabled("LPS-135430")) {
+					putObjectDefinition.setStorageType(StringPool.BLANK);
+				}
 
-		ObjectDefinition putObjectDefinition =
-			objectDefinitionResource.putObjectDefinitionByExternalReferenceCode(
-				objectDefinition.getExternalReferenceCode(), objectDefinition);
+				objectDefinitionResource.putObjectDefinition(
+					putObjectDefinition.getId(), putObjectDefinition);
+			}
+			catch (Exception exception) {
+				if (listImport) {
+					errorsJSONArray.put(
+						JSONUtil.put(
+							"error",
+							_handleImportException(actionRequest, exception)
+						).put(
+							"objectDefinitionName", objectDefinition.getName()
+						));
+				}
+				else {
+					errorsJSONArray.put(
+						_handleImportException(actionRequest, exception));
+				}
+			}
+		}
 
-		putObjectDefinition.setPortlet(objectDefinition.getPortlet());
+		if (errorsJSONArray.length() == 0) {
+			return null;
+		}
 
-		if (!FeatureFlagManagerUtil.isEnabled("LPS-135430")) {
-			putObjectDefinition.setStorageType(StringPool.BLANK);
+		if (!listImport) {
+			return errorsJSONArray.getJSONObject(0);
 		}
 
-		objectDefinitionResource.putObjectDefinition(
-			putObjectDefinition.getId(), putObjectDefinition);
+		return JSONUtil.put(
+			"message", errorsJSONArray
+		).put(
+			"type", "ObjectDefinitionsImportErrorsException"
```

Comment:
@G-Sa the type should be the exception class name, but we don't have the `ObjectDefinitionsImportErrorsException` class.

Reply:
Do we really need this `type`?

Reply:
@G-Sa  we cannot create this exception, as it's not following any pattern. We can try to send the PR with this or try to use a boolean variable.